doc/data-model.md

```diff
@@ -0,0 +1,47 @@
+# Refactoring how and where data is stored
+
+## Background
+
+There are many different kinds of data stored within PMS. At the time of
+writing, they are scattered throughout different parts of the program. As the
+application grows more complex, the internal data representation will need to
+be consolidated.
+
+## Goals
+
+* Create a single entry point for accessing global data.
+* Thread-safe access and manipulation.
+
+## Which data is stored?
+
+The following data collections are shown in list views, directly to the user:
+
+* Track lists
+  * Queue
+  * Library (should be read/write, but undoable)
```

Comment:
The library should be read/write? I didn't know MPD even supported this.

Reply:
MPD doesn't support it. I'm considering a read/write local copy of the library, which can be reverted back to the server copy at any time (i.e. `:e!`), but not written back. The reason is that it might be  confusing for users that different track lists doesn't behave in the same way.